src/pcre2_substitute.c

```diff
@@ -330,18 +742,20 @@ BOOL overflowed = FALSE;
 BOOL use_existing_match;
 BOOL replacement_only;
 BOOL utf = (code->overall_options & PCRE2_UTF) != 0;
-#ifdef SUPPORT_UNICODE
-BOOL ucp = (code->overall_options & PCRE2_UCP) != 0;
-#endif
 PCRE2_UCHAR temp[6];
 PCRE2_SPTR ptr;
 PCRE2_SPTR repend = NULL;
+// XXX AARGH! All the arithmetic on extra_needed is done without overflow
+// checking - it can totally fail, especially on 32-bit systems
```

Comment:
It's not really part of this PR, but I'm tempted to add some overflow checks to the `extra_needed` arithmetic. It's a pre-existing issue. It's just about realistic for a replacement string to have 65,000 copies of "$1", and for $1 to be 65,000 characters long, which would then happily overflow on a 32-bit system.